rust/blockstore/src/arrow/block/delta/single_column_storage.rs

```diff
@@ -250,7 +250,7 @@ impl<V: ArrowWriteableValue<SizeTracker = SingleColumnSizeTracker>> SingleColumn
             )
             .into_inner();
 
-        let mut value_builder = V::get_arrow_builder(inner.size_tracker);
+        let mut value_builder = V::get_arrow_builder(inner.size_tracker.clone());
```

Comment:
This clone could be hairy! What if code makes assumptions about it. Does this really need to take ownership?

Reply:
Yeah can we change this to not take ownership - I know thats out of scope of your PR but this is bug-prone.

Reply:
I think this would largely be fixed with the generic tuple API we've talked about before? e.g. then we wouldn't need specialization like this (I suppose if you wanted to have `Vec<f32> -> FixedSizeList` as an optimization you would still need a separate wrapper trait)